components/script/dom/headers.rs

```diff
@@ -86,6 +86,7 @@ impl HeadersMethods for Headers {
         if let Some(v) = self.header_list.borrow().get_raw(&valid_name) {
             combined_value = v[0].clone();
             combined_value.push(b","[0]);
+            combined_value.push(b" "[0]);
```

Comment:
The underlying `hyper::header::Headers`'s HashMap insert method overwrites entries instead of appending. As a result of this, for a given header name there is at most one value in the inner "header list"/HashMap. I recall that we were debating whether to re-write `hyper::header::Headers` or re-write our `Append` function. We ended up going with the latter. Therefore we ended up combining all the values for a given name together. (I changed the append function to reflect that in [this PR](https://github.com/servo/servo/pull/12634)).